src/deploy/functions/cache/applyHash.ts

```diff
@@ -0,0 +1,46 @@
+import { Backend, allEndpoints } from "../backend";
+import * as args from "../args";
+import {
+  getEndpointHash,
+  getEnvironmentVariablesHash,
+  getSecretsHash,
+  getSourceHash,
+} from "./hash";
+
+/**
+ *
+ * Updates all the CodeBase {@link Backend}, applying a hash to each of their {@link Endpoint}.
+ */
+export async function applyBackendHashToBackends(
+  wantBackends: Record<string, Backend>,
+  context: args.Context
+): Promise<void> {
+  for (const [codebase, wantBackend] of Object.entries(wantBackends)) {
+    const source = context?.sources?.[codebase]; // populated earlier in prepare flow
+    const sourceV1Hash = source?.functionsSourceV1
+      ? await getSourceHash(source?.functionsSourceV1)
+      : undefined;
+    const sourceV2Hash = source?.functionsSourceV2
+      ? await getSourceHash(source?.functionsSourceV2)
+      : undefined;
+    const envHash = getEnvironmentVariablesHash(wantBackend);
+    applyBackendHashToBackend(wantBackend, envHash, sourceV1Hash, sourceV2Hash);
+  }
+}
+
+/**
+ * Updates {@link Backend}, applying a unique hash to each {@link Endpoint}.
+ */
+export function applyBackendHashToBackend(
```

Comment:
nit* Documentation makes me think this function name should be `applyBackendHashToEndpoints` 🤔. But no strong opinions on naming internal api 🤷🏼‍♂️

Reply:
`applyBackendHashToEndpoints` definitely read better. Updated!